### PR TITLE
Fix xss

### PR DESCRIPTION
**Pull request by gjc13**
Rosolve issue #157

### CODE REVIEW

File: src/web/main.cpp [hunk 1]
```diff
@@ -50,6 +50,7 @@
 
 static const char kSyslogIdent[]          = "otWeb";
 static const char kDefaultInterfaceName[] = "wpan0";
+static const char kDefaultListenAddr[]    = "127.0.0.1";
```
Comment (bukepo) [inline src/web/main.cpp]: We should allow request from remote address by default.

File: src/web/main.cpp [hunk 2]
```diff
@@ -103,6 +108,12 @@ int main(int argc, char **argv)
         printf("interfaceName not specified, using default %s\n", interfaceName);
     }
 
+    if (httpListenAddr == NULL)
+    {
+        httpListenAddr = kDefaultListenAddr;
+        printf("litenAddress not specified, using default %s\n", httpListenAddr);
```
Comment (bukepo) [inline src/web/main.cpp]: use `stderr` for error message. typo: `litenAddress` -> `Bind address`.

File: src/web/web-service/web_server.hpp [hunk 3]
```diff
@@ -80,10 +80,11 @@ class WebServer
      * This method starts the Web Server.
      *
      * @param[in]  aIfName  The pointer to the interface name of wpantund.
+     * @param[in]  aListenAddr The http server listen address, can be NULL for any address.
```
Comment (bukepo) [inline src/web/web-service/web_server.hpp]: align the description with above and below lines.

File: src/web/web-service/web_server.cpp [hunk 4]
```diff
@@ -195,7 +199,7 @@ void WebServer::DefaultHttpResponse(void)
 
         } catch (const std::exception &e)
         {
-            std::string content = "Could not open path " + request->path + ": " + e.what();
+            std::string content = "Could not open path";
```
Comment (bukepo) [inline src/web/web-service/web_server.cpp]: how about include the escaped path?

Comment (gjc13) [reply src/web/web-service/web_server.cpp]: Escaping url turns out to be tricky and is not really that necessary

File: src/web/main.cpp (hunk 2 above)
Comment (gjc13) [reply src/web/main.cpp]: It's just a info, so I think stdout suffices.

File: src/web/main.cpp [hunk 5]
```diff
@@ -112,7 +123,7 @@ int main(int argc, char **argv)
     otbrLog(OTBR_LOG_INFO, "border router web started on %s", interfaceName);
 
     server = new ot::Web::WebServer();
-    server->StartWebServer(interfaceName, port);
+    server->StartWebServer(interfaceName, kDefaultListenAddr, port);
```
Comment (bukepo) [inline src/web/main.cpp]: `kDefaultListenAddr` -> `httpListenAddr`?

**Pull request merged**